external/dhcpd/dhcpd_lwip.c

```diff
@@ -103,29 +102,125 @@ typedef struct dhcp_join_data dhcp_join_data_s;
 #define DHCPD_MQ_LEN 11
 #define DHCPD_MQ_MAX_LEN 20
 
-void *_dhcpd_join_handler(void *arg)
+static int _send_mqmsg(mqd_t mq, char *msg, size_t msglen, int prio)
 {
-	dhcp_join_data_s *data = (dhcp_join_data_s *)arg;
-	int ret = ERROR;
-	int sockfd = 0;
-	struct req_lwip_data req;
+	nvdbg("dhcps send dhcps msg size %d\n", msglen);
+	while (1) {
+		int mq_ret = mq_send(mq, msg, msglen, prio);
+		if (mq_ret == ERROR) {
+			if (errno == EAGAIN || errno == EINTR) {
+				continue;
+			}
+			ndbg("dhcps _send_mqmsg error %d\n", errno);
+			return -1;
+		}
+		nvdbg("dhcps sent res %d sent %d\n", mq_ret, msglen);
+		break;
+	}
+	return 0;
+}
+
+static int _recv_mqmsg(mqd_t mq, char *msg, size_t msglen, int *prio)
+{
+	nvdbg("dhcps receive dhcps msg size %d\n", msglen);
+	while (1) {
+		int mq_ret = mq_receive(mq, msg, msglen, prio);
+		if (mq_ret == -1) {
+			if (errno == EAGAIN && errno == EINTR) {
+				continue;
+			}
+			ndbg("dhcps _recv_mqmsg error %d\n", errno);
+			return -1;
+		}
+		if (mq_ret == msglen) {
+			break;
+		} else {
+			ndbg("dhcps critical error %d %d\n", mq_ret, msglen);
+		}
+	}
+	return 0;
+}
+
+static void _empty_mq(mqd_t mq)
+{
+	nvdbg("dhcps empty dhcps message queue\n");
+	char msg[DHCPD_MQ_LEN];
```

Comment:
msg is not initialized.

Reply:
Does buffer have to be initialized before read it?